osu.Game/Online/Chat/ChannelManager.cs

```diff
@@ -348,7 +348,8 @@ public Channel JoinChannel(Channel channel, bool alreadyJoined = false)
                 }
             }
 
-            if (CurrentChannel.Value == null)
+            // If no channel is currently selected, select the newly joined channel
+            if (CurrentChannel.Value == null || CurrentChannel.Value is ChannelSelectorTabItem.ChannelSelectorTabChannel)
```

Comment:
Isn't this going to do the opposite of what we discussed (resulting int he selector closing after each click)?

Reply:
The click is actually being handled elsewhere, this change only affects the joining of the channel on the ChannelManager side.

Reply:
Does that not sound wrong? The ChannelManager is going to have a current channel set that doesn't match the overlay?

Reply:
It does. This is done this way because `ChannelSelectorTabChannel` can actually be set as CurrentChannel.

I guess we should prevent that from happening instead of this change.

Reply:
Well I'm just trying to figure out what you're trying to fix here. Is there an issue showing this happening?

Reply:
If you run the test on master you can see it happening. I can make an issue for that.

Reply:
as discussed, going to prevent `ChannelSelectorTabChannel` from being set as `CurrentChannel` inside `ChannelManager` in the first place